common/lib/common-utils/src/typedEventEmitter.ts

```diff
@@ -58,6 +113,8 @@ export class TypedEventEmitter<TEvent>
         >;
         this.removeListener = super.removeListener.bind(this) as TypedEventTransform<this, TEvent>;
         this.off = super.off.bind(this) as TypedEventTransform<this, TEvent>;
+
```

Comment:
i can only comment on lines with modifications, but the signature of TypedEventEmitter `implements IEventProvider<TEvent & IEvent>` IEvent brings back the string event, and any listener. This was for compatability with exisitng EventEmitters at the time. Here it looks like you intentionally breaking away from EventEmitter, so maybe we don't want that anymore

Reply:
I'm not trying to change how the existing `IEventProvider` functions work.  I think it makes some sense for them to include the fallback, so you can listen to an arbitrary event (e.g. to "cheat" across a compat boundary and listen to an event that's there but not included on whatever interface you have).  We could consider removing that later though (you can still cast to `EventEmitter` or `TypedEventEmitter<IEvent>`)